### ISSUE TITLE
How many API calls are done per one SOQL statement?

### ISSUE THREAD

**Issue by mjcuse** [mike4aday/SwiftlySalesforce#42]
I just started using this today and made it to work. I've been doing some testing today and I think I executed around 100 or so SOQL statements. Tonight, I received an email that the organization has made 11,800+ API calls within the last 24 hours. I wonder if this can be related to with the use of SwiftlySalesforce or maybe something else. Please let me know and any information can help.

**Comment by mike4aday** [comment]
@mjcuse SwiftlySalesforce makes only **1** API call per invocation of the method `query(soql: String) -> Promise<QueryResult>`.

That will return up to 2k records - if there are more records in the query's result set, then the `QueryResult` will have a relative URL in the `nextRecordsPath` property, which you could use as an argument to `queryNext(path: String) -> Promise<QueryResult>`.

What is the size of your SOQL query's result set? Are you calling `queryNext()` until the entire result set is retrieved? Are you testing your app in a 'clean' developer edition org so you could more easily isolate and estimate your app's API consumption? 

I suspect something else is going on.

**Comment by mjcuse** [comment]
Thanks @mike4aday for the quick response. Right after I posted my question, I did some investigation and put breakpoints to my code. I noticed that the SOQL statement was called multiple times for a single transaction. It was on loop. I did some changes to my code so that it will only execute the SOQL once for each transaction. 

Anyway, I am using "salesforce.query(soql: soql).then{queryResult -> () in ..." where soql = "SELECT Name FROM Custom_Table__c WHERE Name = <Unique value>". So it returns 1 record at a time. I am still in the process of making changes to this SQL to return different data from the table. I am just doing existence check for now. Thanks again for your help.
